tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/statements/arrays/ArrayLValueFillTest.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.test.statements.arrays;
+
+import org.ballerinalang.model.values.BMap;
+import org.ballerinalang.model.values.BValueArray;
+import org.ballerinalang.test.util.BCompileUtil;
+import org.ballerinalang.test.util.BRunUtil;
+import org.ballerinalang.test.util.CompileResult;
+import org.ballerinalang.util.exceptions.BLangRuntimeException;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+
+/**
+ * Test cases for filling the elements of a multi-dimensional array when it is used as an LValue.
+ *
+ * @since 1.2.0
+ */
+public class ArrayLValueFillTest {
+
+    private CompileResult compileResult;
+
+    @BeforeClass
+    public void setup() {
+        compileResult = BCompileUtil.compile("test-src/statements/arrays/array_lvalue_fill_test.bal");
+    }
+
+    @Test
+    public void testSimpleBasic2DArrays() {
+        BRunUtil.invoke(compileResult, "testSimpleBasic2DArrays");
+    }
+
+    @Test
+    public void test2DRecordArrays() {
+        BRunUtil.invoke(compileResult, "test2DRecordArrays");
+    }
+
+    @Test
+    public void test2DObjectArrays() {
+        BValueArray arr = (BValueArray) BRunUtil.invokeFunction(compileResult, "test2DObjectArrays")[0];
+
+        assertEquals(arr.size(), 3);
```

Comment:
Can't we write these tests in Ballerina? Maybe in a separate PR?

Reply:
For non-anydata types, there's no direct way to compare values for equality right? That's why I wrote them in Java for objects.